src/libraries/System.Private.Xml/src/System/Xml/XmlQualifiedName.cs

```diff
@@ -36,83 +34,69 @@ public class XmlQualifiedName
         /// </devdoc>
         public XmlQualifiedName(string? name, string? ns)
         {
-            _ns = ns ?? string.Empty;
-            _name = name ?? string.Empty;
+            Namespace = ns ?? string.Empty;
+            Name = name ?? string.Empty;
         }
 
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
-        public string Namespace
-        {
-            get { return _ns; }
-        }
-
+        public string Namespace { get; private set; }
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
-        public string Name
-        {
-            get { return _name; }
-        }
-
+        public string Name { get; private set; }
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
         public override int GetHashCode()
         {
             if (_hash == 0)
             {
-                _hash = Name.GetHashCode() /*+ Namespace.GetHashCode()*/; // for perf reasons we are not taking ns's hashcode.
+                _hash = Name.GetHashCode(); /*+ Namespace.GetHashCode()*/ // for perf reasons we are not taking ns's hashcode.
             }
             return _hash;
         }
 
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
-        public bool IsEmpty
-        {
-            get { return Name.Length == 0 && Namespace.Length == 0; }
-        }
-
+        public bool IsEmpty => Name.Length == 0 && Namespace.Length == 0;
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
         public override string ToString()
         {
-            return Namespace.Length == 0 ? Name : string.Concat(Namespace, ":", Name);
+            return Namespace.Length == 0 ? Name : $"{Namespace}:{Name}";
         }
 
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
         public override bool Equals([NotNullWhen(true)] object? other)
         {
-            if ((object)this == other)
+            if (ReferenceEquals(this, other))
             {
                 return true;
             }
 
-            XmlQualifiedName? qname = other as XmlQualifiedName;
-            if (qname != null)
-            {
-                return (Name == qname.Name && Namespace == qname.Namespace);
-            }
-
-            return false;
+            return other is XmlQualifiedName qName && Name == qName.Name && Namespace == qName.Namespace;
         }
 
         /// <devdoc>
         ///    <para>[To be supplied.]</para>
         /// </devdoc>
         public static bool operator ==(XmlQualifiedName? a, XmlQualifiedName? b)
         {
-            if ((object?)a == (object?)b)
+            if (ReferenceEquals(a, b))
+            {
                 return true;
+            }
 
-            if (a is null || b is null)
+            if (ReferenceEquals(a, null) || ReferenceEquals(b, null))
```

Comment:
use `is null`?

Reply:
https://stackoverflow.com/questions/12465961/is-referenceequalsmyobject-null-better-practice-than-myobject-null

Reply:
`is null` is different from `== null`.  The former is a pure null check, the latter may invoke any operator== overloads on the type.  The use of `is null` was fine here and needn't be replaced by ReferenceEquals.

Reply:
thanks for the confirmation.